napari/layers/shapes/_shapes_models/shape.py

```diff
@@ -240,15 +254,60 @@ def z_index(self):
     def z_index(self, z_index):
         self._z_index = z_index
 
-    def _set_meshes_compiled(
+    def _set_empty_edge(self) -> None:
+        self._edge_vertices = np.empty((0, self.ndisplay), dtype=np.float32)
+        self._edge_offsets = np.empty((0, self.ndisplay), dtype=np.float32)
+        self._edge_triangles = np.empty((0, 3), dtype=np.uint32)
+
+    def _set_empty_face(self) -> None:
+        self._face_vertices = np.empty((0, self.ndisplay), dtype=np.float32)
+        self._face_triangles = np.empty((0, 3), dtype=np.uint32)
+
+    def _set_meshes_compiled_3d(
         self,
         data: CoordinateArray,
         closed: bool = True,
         face: bool = True,
         edge: bool = True,
+    ):
```

Comment:
As I understand it, bermuda special-cases the 3D mesh generation. However, this is *only* called from bermuda (?), so (a) I would rename the function to `_set_meshes_bermuda_3d`, and (b) I would add a docstring "Set meshes when using bermuda backend and data is 3D"

I would also specify in this docstring and in each _set_meshes function that the data is already sliced from nD to the display dimensions, so it can only be 2D or 3D.

Reply:
I have implemented it before you suggest to keep PartSegCore as second compiled backend. 

This is an implementation of your code for searching for common axis and doing triangulation in 2D. 

So in bermuda there is a faster, compiled version, all others are using your code and then python implementation. Should I implement it in PartSegCore also?

Reply:
Not for this PR, just rename in this PR, imho. Let's keep modifications to a minimum for now, as the PR is already a huge endeavour to review.